build/cmake/BoostMacros.cmake

```diff
@@ -38,4 +38,13 @@ macro(REQUIRE_BOOST_LIBRARIES libs)
 	    "or use -DBUILD_TESTING=OFF")
     message(FATAL_ERROR, ${fatal_message})
   endif()
+  if (MSVC)
```

Comment:
These sections are the same.  They are also wrong.
The code that was removed from DefineOptions.cmake was:
```
if (NOT Boost_USE_STATIC_LIBS)
    add_definitions(-DBOOST_ALL_DYN_LINK)
    add_definitions(-DBOOST_TEST_DYN_LINK)
endif()
```

We need to check the link type before assigning any DYN_LINK fields.
I'm also fairly sure these flags do nothing if you are not on Windows, which is why the NO_LIB directive used to be in DefinePlatformOptions...

Reply:
I just moved these lines to the same file with other boost scripts.In fact,I can compile successfully on Windows without these lines,so they could be removed.